rust/azure_iot_operations_protocol/src/rpc_command/executor.rs

```diff
@@ -316,9 +358,37 @@ impl Cache {
     /// `entry` - The cache entry to set.
     fn set(&self, key: CacheKey, entry: CacheEntry) {
         let mut cache = self.0.lock().unwrap();
-        cache.retain(|_, entry| entry.expiration_time.elapsed().is_zero());
+        cache.retain(|_, entry| {
+            match entry {
+                CacheEntry::Cached {
+                    expiration_time, ..
+                } => {
+                    // Retain only non-expired entries
+                    expiration_time.elapsed().is_zero()
+                }
+                CacheEntry::InProgress { .. } => {
+                    // Always retain in progress entries
+                    true
+                }
+            }
+        });
         cache.insert(key, entry);
     }
+
+    /// Remove a cache entry from the cache.
+    fn remove(&self, key: &CacheKey) {
```

Comment:
should there be anything around this or where it's called for this to only be called on InProgress cache entries, or is this fine to happen on Cached entries as well since it would just clean them up slightly sooner?